cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/SourceControlManagementService.java

```diff
@@ -169,4 +192,56 @@ public PushAppResponse pushApp(ApplicationReference appRef, @Nullable String com
 
     return pushResponse;
   }
+
+  /**
+   * Pull the application from linked repository and deploy it in current namespace.
+   * @param appRef application reference to deploy with
+   * @return {@link ApplicationRecord} of the deployed application.
+   * @throws Exception when {@link ApplicationLifecycleService} fails to deploy.
+   */
+  public ApplicationRecord pullAndDeploy(ApplicationReference appRef) throws Exception {
```

Comment:
Throwing `Exception` because the old code `appLifecycleService.deployApp` throws it...

Reply:
we should still have `@throws` in the javadoc for any explicit exceptions that we know about

Reply:
updated javadoc